src/main/java/com/microsoft/azure/functions/worker/reflect/EnhancedClassLoaderProvider.java

```diff
@@ -0,0 +1,76 @@
+package com.microsoft.azure.functions.worker.reflect;
+
+import java.io.*;
+import java.net.*;
+import java.sql.Driver;
+import java.util.*;
+import java.util.concurrent.*;
+
+
+import com.microsoft.azure.functions.worker.*;
+
+public class EnhancedClassLoaderProvider implements ClassLoaderProvider {
+    public EnhancedClassLoaderProvider() {
+        urls = Collections.newSetFromMap(new ConcurrentHashMap<URL, Boolean>());
+    }
+
+    /*
+     * @see com.microsoft.azure.functions.reflect.ClassLoaderProvider#getClassLoader()
+     */
+    @Override
+    public ClassLoader getClassLoader() {
+        URL[] urlsForClassLoader = new URL[urls.size()];
+        urls.toArray(urlsForClassLoader);
+
+        URLClassLoader classLoader = new URLClassLoader(urlsForClassLoader);
+        loadDrivers(classLoader);

```

Comment:
You are loading the drivers here as a side effect of calling getClassLoader. I would recommend where you call getClassLoader to be followed by the code from loadDrivers

Reply:
I updated the name as we discussed to createClassLoader.